ui/src/components/WMWorkerPoolsTable/index.jsx

```diff
@@ -112,36 +118,87 @@ export default class WorkerManagerWorkerPoolsTable extends Component {
     this.setState({ sortBy: header.id, sortDirection });
   };
 
-  handleDeleteClick = async ({ currentTarget: { name } }) => {
-    const workerPool = this.props.workerPools.find(
-      wp => wp.workerPoolId === name
-    );
+  handleDeleteClick = async () => {
+    const { item } = this.state.dialogState;
     const payload = {
-      providerId: workerPool.providerId,
-      description: workerPool.description,
-      config: workerPool.config,
-      owner: workerPool.owner,
-      emailOnError: workerPool.emailOnError,
+      providerId: item.providerId,
+      description: item.description,
+      config: item.config,
+      owner: item.owner,
+      emailOnError: item.emailOnError,
     };
 
     this.props.history.replace('/worker-manager');
 
     try {
       await this.props.deleteRequest({
-        workerPoolId: workerPool.workerPoolId,
+        workerPoolId: item.workerPoolId,
         payload,
       });
+      this.setState({
+        dialogState: {
+          error: null,
+          open: false,
+          title: '',
+          body: '',
+          confirmText: '',
+          item: null,
+        },
+      });
     } catch (error) {
-      this.setState({ error: formatError(error), actionLoading: false });
+      this.handleDialogActionError(error);
     }
   };
 
+  handleDialogActionOpen = workerPool => {
+    this.setState({
+      dialogState: {
+        error: null,
+        open: true,
+        title: 'Delete Worker Pool?',
+        body: `This will delete the worker pool ${workerPool.workerPoolId}.`,
+        confirmText: 'Delete Worker Pool',
+        item: workerPool,
+      },
+    });
+  };
+
+  handleDialogActionError = error => {
+    const { item } = this.state.dialogState;
+
+    this.setState({
+      dialogState: {
+        ...this.state.dialogState,
+        error,
+        body: `Could not delete ${item.workerPoolId}. Please try again.`,
+      },
+    });
+  };
+
+  handleDialogActionComplete = () => {
+    this.props.history.push('/worker-manager');
+  };
+
+  handleDialogActionClose = () => {
+    this.setState({
+      dialogState: {
+        error: null,
+        open: false,
+        title: '',
+        body: '',
+        confirmText: '',
+        item: null,
+      },
+    });
```

Comment:
No need to remove the title, body, confirmText, item when closing the dialog. We can just set `error: null, open: false`. Also, since we're setting `error: null` here, then no need to set `error: null` in `handleDialogActionOpen`.